dev/io.openliberty.concurrent.internal.cdi/src/io/openliberty/concurrent/internal/cdi/ManagedThreadFactoryBean.java

```diff
@@ -124,6 +113,46 @@ public class ManagedThreadFactoryBean implements Bean<ManagedThreadFactory>, Pas
         }
     }
 
+    private ClassLoader findApplicationClassloader(J2EEName j2eeName) {
+        BundleContext bc = FrameworkUtil.getBundle(ClassLoaderIdentifierService.class).getBundleContext();
+
+        if (Objects.isNull(bc)) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "No bundle context found for ClassLoaderIdentifierService "
+                             + "cannot configure a classloader for the default ManagedThreadFactory bean");
+            }
+            // TODO throw exception instead?
+            return null;
+        }
+
+        ServiceReference<ClassLoaderIdentifierService> ref = bc.getServiceReference(ClassLoaderIdentifierService.class);
+        try {
+            ClassLoaderIdentifierService classloaderIdSvc = bc.getService(ref);
```

Comment:
Only invoke `ungetService` if `getService` is successful
```suggestion
        ClassLoaderIdentifierService classloaderIdSvc = bc.getService(ref);
        try {
```

Reply:
I don't think this one applies anymore. The code changed while I was reviewing it.